Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/ContentLocationEventData.cs

```diff
@@ -310,6 +318,55 @@ public override int GetHashCode()
         {
             return (base.GetHashCode(), AccessTime.GetHashCode()).GetHashCode();
         }
+    }
+
+    /// <nodoc />
+    public sealed class AddContentLocationWithoutTouchingEventData : ContentLocationEventData
+    {
+        /// <summary>
+        /// A list of content sizes associated with <see cref="ContentLocationEventData.ContentHashes"/>.
+        /// </summary>
+        public IReadOnlyList<long> ContentSizes { get; }
+
+        /// <nodoc />
+        public AddContentLocationWithoutTouchingEventData(MachineId sender, IReadOnlyList<ShortHashWithSize> addedContent)
+            : this(sender, addedContent.SelectList(c => c.Hash), addedContent.SelectList(c => c.Size))

```

Comment:
This is probably too much to undertake now, but not a fan of doing all this list allocation just to make calls work. Can we refactor use co-variant list types or iterators?